plugins/modules/ndo_l3out_bgp_peer.py

```diff
@@ -0,0 +1,800 @@
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+module: ndo_l3out_bgp_peer
+short_description: Manage L3Out BGP Peer on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage L3Out BGP Peer on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Sabari Jaganathan (@sajagana)
+options:
+  template:
+    description:
+    - The name of the L3Out template.
+    type: str
+    aliases: [ l3out_template ]
+  template_id:
+    description:
+    - The ID of the L3Out template.
+    type: str
+    aliases: [ l3out_template_id ]
+  l3out:
+    description:
+    - The name of the L3Out.
+    type: str
+  l3out_uuid:
+    description:
+    - The UUID of the L3Out.
+    type: str
+  node_group:
+    description:
+    - The name of the Node Group Policy.
+    type: str
+    required: true
+  ipv4_address:
+    description:
+    - The IPv4 address of the L3Out BGP Peer.
+    - Providing an empty string will remove the O(ipv4_address="") from the L3Out BGP Peer.
+    type: str
+    aliases: [ peer_address_ipv4 ]
+  ipv6_address:
+    description:
+    - The IPv6 address of the L3Out BGP Peer.
+    - Providing an empty string will remove the O(ipv6_address="") from the L3Out BGP Peer.
+    type: str
+    aliases: [ peer_address_ipv6 ]
+  remote_asn:
+    description:
+    - The remote autonomous system number (ASN) of the L3Out BGP Peer.
+    - The value must be between 1 and 4294967295.
+    type: int
+  admin_state:
+    description:
+    - The administrative state of the L3Out BGP Peer.
+    - Defaults to C(enabled) when unset during creation.
+    type: str
+    choices: [ enabled, disabled ]
+  auth_password:
+    description:
+    - The authentication password of the L3Out BGP Peer.
+    type: str
+  ebgp_multi_hop_ttl:
+    description:
+    - The TTL for eBGP multi-hop of the L3Out BGP Peer.
+    - Defaults to 1 when unset during creation.
+    - The value must be between 1 and 255.
+    type: int
+  site_of_origin:
+    description:
+    - The site of origin for the L3Out BGP Peer. The value must adhere to the pattern "extended:as2-nn2:1000:65534".
+    type: str
+    aliases: [ fabric_of_origin ]
+  weight:
+    description:
+    - The weight of the L3Out BGP Peer.
+    - The value must be between 1 and 65535.
+    type: int
+  allowed_self_as_count:
+    description:
+    - The allowed self-AS count of the L3Out BGP Peer.
+    - Defaults to 3 when unset during creation.
+    - The value must be between 1 and 10.
+    type: int
+  local_asn_config:
+    description:
+    - The configuration for local ASN of the L3Out BGP Peer.
+    type: str
+    choices: [ none, no_prepend, dual_as, replace_as ]
+  local_asn:
+    description:
+    - The local autonomous system number (ASN) of the L3Out BGP Peer.
+    - The value must be between 1 and 4294967295.
+    type: int
+  import_route_map:
```

Comment:
Should the `route_map` and `peer_prefix` parameters be a dictionary with `name` and `template` options? How do we specify which template the objects are from?

Reply:
The `route_map` objects are unique within the tenant level, so we can skip the template check for the `route_map` objects.
The `peer_prefix` also unique within the tenant level, but it allows to import common tenant `peer_prefix` value. Let me make changes for both route map and peer prefix.

Reply:
The `route_map` objects are unique within the tenant level, so we can skip the template check for the `route_map` objects.
The `peer_prefix` also unique within the tenant level, but it allows to import common tenant `peer_prefix` value. Let me make changes for both route map and peer prefix.

Reply:
see choices made in this discussion where it is a map with name, https://github.com/CiscoDevNet/ansible-mso/pull/577/files#r1970341772. perhaps we should follow same here?